### DISCUSSION TITLE
the default PATH value for macOS silicion

### DISCUSSION THREAD

**Discussion by kwenlyou** [Homebrew/discussions#2237] [Everyday usage] [answered]
I have a quick question regarding the default `PATH` value set by homebrew for macOS silicon. 

After the installation of homebrew, following the advice, I added `eval "$(brew shellenv)"` in `~/.zprofile`. However, I noticed `/opt/homebrew/bin` is after `/usr/bin` as follows
```
PATH= /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/opt/homebrew/sbin
```

Is this on purpose for some subtle reason?

**Comment by carlocab** [comment, marked as answer]
That’s not on purpose. Something else in your configuration is changing your `PATH`, because `brew shellenv` makes sure `/opt/homebrew/bin` comes before other entries.

**Comment by kwenlyou** [comment]
@carlocab Thanks. Uninstall homebrew and install it again. The `PATH` is right now. Not sure what happened.

**Comment by tjluoma** [comment]
on macOS, `zsh` gets its default `$PATH` from `/etc/zprofile`.

If you want to make sure that it is set properly the way you want it, be sure to set it in `~/.zshenv`

**Comment by kwenlyou** [comment]
@tjluoma thanks. It was tmux messed up `PATH` due to https://superuser.com/questions/544989/does-tmux-sort-the-path-variable

**Reply by serl** [reply]
I had the same issue on bash (wrong PATH order in login sub-shells, like tmux), the solution in https://superuser.com/questions/544989/does-tmux-sort-the-path-variable totally fixed my issue, thanks @kwunlyou! 

For the ones who don't want to do the extra click, that basically boils down to prepending to your `~/.bash_profile`:

```sh
PATH=""
source /etc/profile
```

(followed by `brew shellenv` and all the rest)
